src/identity.cc

```diff
@@ -275,17 +369,24 @@ class ModelInstanceState {
   // Get the state of the model that corresponds to this instance.
   ModelState* StateForModel() const { return model_state_; }
 
+  // Returns the stream associated with this instance that can be used
+  // for GPU<->CPU memory transfers. Returns nullptr if GPU support is
+  // disabled or if this instance is not executing on a GPU.
+  cudaStream_t CudaStream() { return stream_; }
```

Comment:
You can get this for free if you inherit from `BackendModelInstance` class

Reply:
And the create / destroy below is also handled.

Reply:
We have extra components and restrictions in BackendModelInstance and BackendModel (for ModelState) that we do not need to impose here.